### PR TITLE
make Pool's `CLIENT_SEARCH_SPACE_BYTES` more permissive

### PR DESCRIPTION
**Pull request by plebhash**
close stratum-mining/sv2-apps#44

since tProxy will likely operate with `downstream_extranonce2_size` at most 8, allowing total of 16 bytes for Pool's clients should be sufficient for most cases

### CODE REVIEW

File: roles/pool/src/lib/channel_manager/mod.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ mod mining_message_handler;
 mod template_distribution_message_handler;
 
 const POOL_ALLOCATION_BYTES: usize = 4;
-const CLIENT_SEARCH_SPACE_BYTES: usize = 8;
+const CLIENT_SEARCH_SPACE_BYTES: usize = 16;
```
Comment (bit-aloo) [inline roles/pool/src/lib/channel_manager/mod.rs:49]: Should we make it 28?

Comment (plebhash) [reply roles/pool/src/lib/channel_manager/mod.rs:49]: there's no fundamental reason preventing us from doing that

however, that's highly suggestive of the misconception that full extranonces must be fixed at 32 bytes, which in turn makes us highly susceptible to blindspots

in fact, I foresee some challenges around this topic when we start making adaptations coming from https://github.com/stratum-mining/sv2-spec/pull/162

so I'd like to avoid that

Comment (bit-aloo) [reply roles/pool/src/lib/channel_manager/mod.rs:49]: Ah cool!!! let me check the spec stuff. Consider it non-blocking.

File: test/integration-tests/tests/jd_tproxy_integration.rs [hunk 2]
```diff
@@ -2,15 +2,61 @@ use integration_tests_sv2::{interceptor::MessageDirection, template_provider::Di
...
+            MessageDirection::ToDownstream,
+            MESSAGE_TYPE_OPEN_EXTENDED_MINING_CHANNEL_SUCCESS,
+        )
+        .await;
+    jdc_pool_sniffer
+        .wait_for_message_type(
+            MessageDirection::ToUpstream,
+            MESSAGE_TYPE_SUBMIT_SHARES_EXTENDED,
+        )
+        .await;
```
Comment (bit-aloo) [inline test/integration-tests/tests/jd_tproxy_integration.rs:62]: This test doesn’t actually verify the behavior in non-aggregated mode. Ideally, we should check whether multiple mining instance connections result in multiple OpenExtendedMiningChannel messages.

Comment (plebhash) [reply test/integration-tests/tests/jd_tproxy_integration.rs:62]: how can we achieve that?

unfortunately asserting for two identical messages is not trivial...

if we check the inner workings of `wait_for_message_type`, we'll see that it's redundant to call it 2x for the same message type

I tried doing `wait_for_message_type_and_clean_queue` but the test started timing out

---

I see how the test is asserting less than ideal

but overall, my goal with adding this test was just to make sure the original edge case that motivated this PR would be caught on CI

if we keep the test and revert the change on pool, it should crash

Comment (plebhash) [reply test/integration-tests/tests/jd_tproxy_integration.rs:62]: wait, just realized I was doing some dumb stuff

Comment (plebhash) [reply test/integration-tests/tests/jd_tproxy_integration.rs:62]: done
